src/plugin/plugin.cpp

```diff
@@ -83,6 +85,8 @@ class Q_DECL_EXPORT NemoEmailPlugin : public QQmlExtensionPlugin
         qmlRegisterType<AttachmentListModel>(uri, 0, 1, "AttachmentListModel");
         qmlRegisterUncreatableType<FolderAccessor>(uri, 0, 1, "FolderAccessor",
                                                    "FolderAccessor is created via FolderListModel or similar");
+        qmlRegisterType<EmailContactListModel>(uri, 0, 1, "EmailContactListModel");
+        qmlRegisterType<EmailContactListProxyModel>(uri, 0, 1, "EmailContactListProxyModel");
```

Comment:
Commented already earlier on the app side, but this proxy model thing is a bit strange as the proxied source model is set internally. For the app developer the proxy part is almost invisible. Only funkier detail could be that the proxy model is a property in the qt class so that exposes the internals outside.

Then the EmailContactListModel itself is not at the moment used at all on the app side? Does it need to be registered and exposed? Is there need to even have a separate proxy and source or could it all be embedded into a single contact model?

Reply:
The separation into model and proxy model is needed.
They have different responsibilities and tasks.
Now there is no need for a model outside, fixed it.

Reply:
proxy are easier to implement in C++ side.
In QML side model name are EmailContactModel  (remove suffix List)